pkg/reduce/readloop/readloop.go

```diff
@@ -150,8 +152,15 @@ func (rl *ReadLoop) Process(ctx context.Context, messages []*isb.ReadMessage) {
 	// for each successfully written message,
 	for _, m := range successfullyWrittenMessages {
 		// close any windows that need to be closed.
+		var closedWindows []window.AlignedKeyedWindower
 		wm := processor.Watermark(m.Watermark)
-		closedWindows := rl.windower.RemoveWindows(time.Time(wm))
+
+		// check to see if any windows can be closed, when the watermark is moved
+		if wm.After(time.Time(rl.latestWatermark)) {
```

Comment:
Why are we comparing 2 watermarks? Isn't it obvious that those 2 will be diverging (the old one and the new one)?

Reply:
they can be same, I just wanted to invoke the remove windows function only when the watermark is progressed